src/interfaces/diagram.ts

```diff
@@ -1,18 +1,56 @@
+export interface IDiagram {
+  tablePosition: ITablePosition[];
+  relationPosition: IRelationOnDiagram[];
+}
+
+export interface ITablePosition {
+  name: string;
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+}
+
+export interface IRelationOnDiagram {
+  name: string;
```

Comment:
as discussed should have "from" and "to" of type 
IRelativePoint { side: SideEnum, shift: number }

type Side enum = {
 left: "l",
 right: "r",
 top: "t",
 bottom: "b"
}